schema/changelog-3.15.xml

```diff
@@ -83,5 +83,54 @@
       <column name="limitcommands" type="BOOLEAN" defaultValueBoolean="false" />
     </addColumn>
 
+    <addColumn tableName="notifications">
+      <column name="allways" type="BOOLEAN" defaultValueBoolean="false" valueBoolean="true">
+        <constraints nullable="false" />
+      </column>
+    </addColumn>
+
+    <createTable tableName="user_notification">
+      <column name="userid" type="INT">
+        <constraints nullable="false" />
+      </column>
+      <column name="notificationid" type="INT">
+        <constraints nullable="false" />
+      </column>
+    </createTable>
+
+    <addForeignKeyConstraint baseTableName="user_notification" baseColumnNames="userid" constraintName="fk_user_notification_userid" referencedTableName="users" referencedColumnNames="id" onDelete="CASCADE" />
+    <addForeignKeyConstraint baseTableName="user_notification" baseColumnNames="notificationid" constraintName="fk_user_notification_notificationid" referencedTableName="notifications" referencedColumnNames="id" onDelete="CASCADE" />
+
+    <sql>
+      INSERT INTO user_notification (notificationid, userid) SELECT id AS notificationid, userid FROM notifications;
```

Comment:
Migrate old notifications to new scheme transparent.

Reply:
By the way, have you tested this on different databases? I think it should work, but I'm not completely sure. I think we should test at least H2, MySQL and MS SQL.